hearth/lib/hearth/validator.rb

```diff
@@ -41,5 +41,19 @@ def self.validate_unknown!(struct, params, context:)
       raise ArgumentError,
             "Unexpected members: [#{unknown.join(', ')}]"
     end
+
+    # Validate the given value is within the expected range (inclusive).
+    # @param value [Object] The value to validate.
+    # @param min [Integer] The minimum that the given value should be.
```

Comment:
I think this should work for any `Numeric`, so I think we could update the doc type here.  (Our RangeConstraint in Java codegen currently only applies for integer types, but this code is more flexible).